src/Storefront/Theme/ThemeRuntimeConfig.php

```diff
@@ -0,0 +1,83 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Storefront\Theme;
+
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * @internal
+ *
+ * DTO to store precalculated configuration of a theme used during storefront rendering.
+ * Used to avoid recalculating the configuration for every request.
+ *
+ * Most of the properties are calculated during Shopware\Storefront\Theme\ThemeLifecycleService::refreshTheme.
+ * The $scriptFiles are calculated just after Shopware\Storefront\Theme\ThemeCompiler::compileTheme.
+ */
+#[Package('framework')]
+class ThemeRuntimeConfig
+{
+    public function __construct(
+        public readonly string $themeId,
+        public readonly ?string $technicalName,
+        /** @var array<string, mixed> */
+        public readonly array $resolvedConfig,
+        /** @var array<string> */
+        public readonly array $viewInheritance,
+        /** @var array<string>|null */
+        public readonly ?array $scriptFiles,
+        /** @var array<string, array{path: string, namespace: string}> */
+        public readonly array $iconSets,
+        public readonly \DateTimeInterface $updatedAt
+    ) {
+    }
+
+    /**
+     * @param array{
```

Comment:
does it make sense to add this as a local type alias? so we don't have to redefine the same type information all over the place? https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases

I expect this is used more often, as that is also the structure we get back from the DB right?